app/parse_txs.go

```diff
@@ -1,105 +1,69 @@
 package app
 
 import (
-	"crypto/sha256"
-	"errors"
-
-	"github.com/celestiaorg/celestia-app/app/encoding"
-	"github.com/celestiaorg/celestia-app/x/blob/types"
+	blobtypes "github.com/celestiaorg/celestia-app/x/blob/types"
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/x/auth/signing"
-	core "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/libs/log"
 	coretypes "github.com/tendermint/tendermint/types"
 )
 
 // parsedTx is an internal struct that keeps track of potentially valid txs and
 // their wire messages if they have any.
 type parsedTx struct {
```

Comment:
[optional] two helpers that may make usage of parsedTx slightly more readable

```go
func (p parsedTx) isNormalTx() bool {
	return len(p.normalTx) != 0
}

func (p parsedTx) isBlobTx() bool {
	return len(p.blobTx.Tx) != 0
}
```

Reply:
cool, and yeah good idea, added for a future PR.